### ISSUE TITLE
Model Performance is Suboptimal

### ISSUE THREAD

**Issue by Oufattole** [Oufattole/meds-torch#39]
I observed the training loss (for both long_los and icu_mortality tasks in the [MIMIC IV tutorial](https://meds-torch.readthedocs.io/en/latest/tutorial/)) is unstable and nonconverging for supervised training. Even after tuning the loss just oscillates over epochs.

I added a synthetic data test in the [tune_debug branch](https://github.com/Oufattole/meds-torch/tree/tune_debug) that should be easily learnable, the model only has to learn whether or not there is the code "ADMISSION//CARDIAC" or "ADMISSION//ONCOLOGY" in the patient's history. And I trained an LSTM on this and get these metrics over $1000$ epochs and with learning rate $0.1$:
![image](https://github.com/user-attachments/assets/15fc5c75-077a-41b8-87e3-538c2862d3b9)
![image](https://github.com/user-attachments/assets/bd0aac80-c6a8-4661-96b5-5f6e6eeec6c0)

I ran this with the command:
```bash
pytest -k test_model_memorization_train[pytorch_dataset-supervised-earlyfusionNone-triplet_encoder-lstm] -s
```

I also trained a transformer with the same data and task and got these (note that the learning rate is in the legend for the three learning rates I tried):
![image](https://github.com/user-attachments/assets/182c8e49-41ac-466e-8bd3-f65ca94ac085)
![image](https://github.com/user-attachments/assets/39babb10-4c92-4c67-9421-2a0aba8c956d)

The command used is:
```bash
pytest -k test_model_memorization_train[pytorch_dataset-supervised-earlyfusionNone-triplet_encoder-transformer_encoder] -s
```

**Comment by Oufattole** [comment]
I tried training the transformer encoder with an extremely low learning rate (1e-16). I would expect to get very little changes in metrics over epochs for such a small learning rate, and validation metrics follow this, but training loss is still oscillating:

![image](https://github.com/user-attachments/assets/33b4d981-6821-4b0c-955c-977b81c7ab3c)
![image](https://github.com/user-attachments/assets/efcb0216-a88e-4c46-8efc-a9c5767280fa)
